hydrus/helpers.py

```diff
@@ -491,3 +488,49 @@ def send_update(method: str, path: str):
     last_job_id = crud.get_last_modification_job_id(session)
     new_job_id = crud.insert_modification_record(method, resource_url, session)
     send_sync_update(socketio, new_job_id, last_job_id, method, resource_url)
+
+
+def get_type_from_path(path: str) -> str:
+    _, parsed_classes = get_collections_and_parsed_classes()
+    expanded_base_url = DocUrl.doc_url
+    for class_name in parsed_classes:
+        class_ = parsed_classes[class_name]['class']
+        if path == class_.path:
+            return class_.id_.split(expanded_base_url)[1]
+
+
+def get_path_from_type(type_: str) -> str:
+    _, parsed_classes = get_collections_and_parsed_classes()
+    expanded_base_url = DocUrl.doc_url
+    for class_name in parsed_classes:
+        class_ = parsed_classes[class_name]['class']
+        if type_ == class_.id_.split(expanded_base_url)[1]:
+            return class_.path
+
+
+def parse_collection_members(object_: dict) -> dict:
+    """Parse the members of a collection to make it easier
+    to insert in database.
+
+    :param object_: The body of the request having object members
+    :type object_: dict
+    :return: Object with parsed members
+    :rtype: dict
+    """
+    members = list()
+    for member in object_['members']:
+        # eg member "/serverapi/LogEntry/7b51ac78-e917-4884-b52e-04e65d19b810"
+        member_components = member.split('/')
+        member_id = member_components[-1]
+        member_path = member_components[-2]
+        member_type = get_type_from_path(member_path)
+        if crud.item_exists(member_type, member_id, get_session()):
```

Comment:
Where are we checking if the member is managed by the collection or not?

Reply:
Currently, we are **not** checking if the member is managed by the collection or not. I did this on purpose as on reading the [spec](https://www.hydra-cg.com/spec/latest/core/#manages-block), it was written that "a `manages` node to the collection instructs the **client**".

So, I assumed that the `manages` block is instruction for the client rather than the server and therefore the overhead of checking if the member is managed by the collection or not falls on the client.

Let me know if we need to check the manages block at the server also?